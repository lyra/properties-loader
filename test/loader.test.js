import compiler from './compiler.js'

function getOutput(stats) {
  return stats.toJson({ source: true }).modules[0].source
}

describe('Load a properties file and outputs a JSON module', () => {
  test('Basic example', async () => {
    const stats = await compiler('example-1.properties')
    expect(getOutput(stats)).toMatchSnapshot()
  })

  test('Example with namespaces', async () => {
    const stats = await compiler('example-2.properties', { namespaces: true })
    expect(getOutput(stats)).toMatchSnapshot()
  })

  test('Example with namespaces and sections', async () => {
    const stats = await compiler('example-3.properties', { namespaces: true, sections: true })
    expect(getOutput(stats)).toMatchSnapshot()
  })
})
