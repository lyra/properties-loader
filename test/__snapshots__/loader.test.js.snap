// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load a properties file and outputs a JSON module Basic example 1`] = `"export default {\\"name\\":\\"alex\\"}"`;

exports[`Load a properties file and outputs a JSON module Example with namespaces 1`] = `"export default {\\"user\\":{\\"name\\":\\"alex\\"}}"`;

exports[`Load a properties file and outputs a JSON module Example with namespaces and sections 1`] = `"export default {\\"user\\":{\\"name\\":\\"alex\\"},\\"address\\":{\\"number\\":42,\\"street\\":\\"foobar\\"}}"`;
